pandas/tests/io/parser/test_na_values.py

```diff
@@ -812,3 +812,32 @@ def test_bool_and_nan_to_float(all_parsers):
     result = parser.read_csv(StringIO(data), dtype="float")
     expected = DataFrame.from_dict({"0": [np.nan, 1.0, 0.0]})
     tm.assert_frame_equal(result, expected)
+
+
+@xfail_pyarrow
+@pytest.mark.parametrize(
+    "na_values, expected_result, test_id",
+    [
+        (
+            {"A": [-99.0, -99]},
+            DataFrame({"A": [np.nan, np.nan, np.nan, np.nan]}),
```

Comment:
Could you move this to the body of the test since it's the same for both parameterized cases?